sos/report/plugins/gluster.py

```diff
@@ -41,15 +41,16 @@ def wait_for_statedump(self, name_dir):
                     ret = string.count(last_line, 'DUMP_END_TIME')
 
     def postproc(self):
-        if not os.path.exists(self.statedump_dir):
-            return
-        try:
-            for dirs in os.listdir(self.statedump_dir):
-                os.remove(os.path.join(self.statedump_dir, dirs))
-            os.rmdir(self.statedump_dir)
-            os.unlink('/tmp/glusterdump.options')
-        except OSError:
-            pass
+        if self.get_option("dump"):
+            if not os.path.exists(self.statedump_dir):
+                return
+            try:
+                for name in os.listdir(self.statedump_dir):
+                    if 'dump' in name:
+                        os.remove(os.path.join(self.statedump_dir, name))
+                os.unlink('/tmp/glusterdump.options')
```

Comment:
This is a left-over from the patch that introduced the regression:

```diff
commit 7b485f02c39b0d8502c224dd90c27024951948dc
Author: Jake Hunsaker <jhunsake@redhat.com>
Date:   Wed Apr 17 17:00:03 2019 -0400

    [gluster] Add volume, state information and dump glusterd processes
[...]
@@ -40,17 +40,6 @@ class Gluster(Plugin, RedHatPlugin):
         fp.close()
         return out
 
-    def make_preparations(self, name_dir):
-        try:
-            os.mkdir(name_dir)
-        except OSError:
-            pass
-        fp = open('/tmp/glusterdump.options', 'w')
-        data = 'path=' + name_dir + '\n'
-        fp.write(data)
-        fp.write('all=yes')
-        fp.close()
-
```

All the code that generated this file was already removed so we should get rid of this as well.

Reply:
os.unlink('/tmp/glusterdump.options') removed